gaitmesh_ros/include/gaitmesh_ros/common.hpp

```diff
@@ -26,5 +32,17 @@ inline bool pointsOnSameSideOfLine(const Eigen::Vector3d& p1, const Eigen::Vecto
 
 inline bool pointInTriangle(const Eigen::Vector3d& p, const Eigen::Vector3d& a, const Eigen::Vector3d& b, const Eigen::Vector3d& c)
 {
+    if ((p - a).norm() < 1e-3 || (p - b).norm() < 1e-3 || (p - c).norm() < 1e-3)
+        return true;
     return pointsOnSameSideOfLine(p, a, b, c) && pointsOnSameSideOfLine(p, b, a, c) && pointsOnSameSideOfLine(p, c, a, b);
 }
+
+inline double slope(const Eigen::Vector3d& n)
+{
+    Eigen::Vector3d k(0, 0, 1);
+    if (n.dot(k) < 1e-4 && n.dot(k) >= 0)
```

Comment:
Compute `n.dot(k)` in a temporary variable outside - since it's used a few times here...